allowed.py

```diff
@@ -640,7 +659,7 @@ def check_file(filename: str, constructs: tuple) -> None:
     for name in args.file_or_folder:
         if os.path.isdir(name):
             check_folder(name, args.unit)
-        elif name.endswith(".py"):
+        elif is_filetype_supported(name):
```

Comment:
It's simpler and faster to check if name ends with py or ends with ipynb than calling a function that splits the pathname.

Reply:
That's fair enough. I switched for the function as I thought it read better than `elif name.endswith("py") or name.endswith("ipynb")` but didn't really consider the performance implicateions. Have switched back.